org.jacoco.core/src/org/jacoco/core/internal/analysis/filter/SyntheticFilter.java

```diff
@@ -21,11 +21,22 @@
 
 	public void filter(final MethodNode methodNode,
 			final IFilterContext context, final IFilterOutput output) {
-		if ((methodNode.access & Opcodes.ACC_SYNTHETIC) != 0
-				&& !methodNode.name.startsWith("lambda$")) {
-			output.ignore(methodNode.instructions.getFirst(),
-					methodNode.instructions.getLast());
+		if ((methodNode.access & Opcodes.ACC_SYNTHETIC) == 0) {
+			return;
 		}
+
+		if (methodNode.name.startsWith("lambda$")) {
+			return;
+		}
+
+		if (methodNode.name.endsWith(KotlinDefaultArgumentsFilter.SUFFIX)
```

Comment:
Instead of duplicating the logic here I would prefer to encapsulate it in the respective filters, i.e.

* `KotlinGeneratedFilter.isKotlinClass(IFilterContext cts)` and
* `KotlinDefaultArgumentsFilter.isDefaultArgumentMethod(String name)`